budgeteer-web-interface/src/main/java/org/wickedsource/budgeteer/web/BudgeteerSession.java

```diff
@@ -61,8 +63,8 @@ public BudgetTagFilter getBudgetFilter() {
         return budgetFilter.get(getProjectId());
     }
 
-    public void setBudgetFilter(BudgetTagFilter budgetFilter) {
-        this.budgetFilter.put(projectId, budgetFilter);
+    public void setBudgetFilter(BudgetTagFilter budgetTagFilter) {
```

Comment:
It now would make sense to rename this method to empathize it is using the *tag* filter. Not your job in this PR, but something to keep in mind